tcms/rpc/tests/test_bug.py

```diff
@@ -0,0 +1,29 @@
+import unittest
+
+from django.conf import settings
+from django.core.cache import cache
+
+from tcms.rpc.tests.utils import APITestCase
+
+if 'tcms.bugs.apps.AppConfig' not in settings.INSTALLED_APPS:
+    raise unittest.SkipTest('tcms.bugs is disabled')
+
+
+class TestBug(APITestCase):
+
+    def test_get_details_from_cache(self):
+        url = "http://some.url"
+        expected_result = {
+            'title': 'Bug from cache',
+            'description': 'This bug came from the Django cache'
+        }
+        cache.set(url, expected_result)
+        result = self.rpc_client.Bug.details(url)
+
+        self.assertEqual(result, expected_result)
+
+    def test_empty_details_when_tracker_does_not_exist(self):
+        url = "http://unknown-tracker.url"
```

Comment:
If the tracker is unknown then `tracker_from_url()` will return None which should then break when calling `.details()` on it.

One possible way to deal with this is configure a known bug tracker system (e.g. Kiwi TCMS itself) and use it to validate that the API will return what's expected. 

Another way is to assert on mock objects that the API is calling the tracker integration code and rely on the other tests which exercise the tracker integration code directly.

Reply:
if the tracker is unknown the API will return `{}` and this is the functionality that is tested here

Reply:
^^^ this is a test scenario for the refactored version of the method and you need to first have a test for the method version in master, that test to be passing and then we expect it to be passing as well after the refactoring. The original implementation doesn't have this behavior so I guess this needs to be a new test after the refactoring takes place. 

It may be easier to first work only on the tests for the code in master and then come back and rebase the refactoring + new tests on top of that.

Reply:
I reverted the refactoring, and ignore the unit tests. let's merge this and will add the refactoring in another PR

Reply:
This test method doesn't belong here. Move it together with the other PR.